modules/core/keeper/keeper.go

```diff
@@ -50,12 +50,28 @@ func NewKeeper(
 	}
 
 	// panic if any of the keepers passed in is empty
-	if reflect.ValueOf(stakingKeeper).IsZero() {
-		panic(fmt.Errorf("cannot initialize IBC keeper: empty staking keeper"))
+
+	// switch case to detect blank pointers
+	switch reflect.TypeOf(stakingKeeper).Kind() {
+	case reflect.Ptr:
+		if reflect.ValueOf(&stakingKeeper).IsZero() {
+			panic(fmt.Errorf("cannot initialize IBC keeper: empty staking keeper"))
+		}
+	default:
+		if reflect.ValueOf(stakingKeeper).IsZero() {
+			panic(fmt.Errorf("cannot initialize IBC keeper: empty staking keeper"))
```

Comment:
Could we maybe have a slight different error message, so that in case the error occurs is immediately obvious what the problem was (i.e. either empty value or empty pointer)?
```suggestion
			panic(fmt.Errorf("cannot initialize IBC keeper: empty staking keeper value"))
```